hw2/03-count.js

```diff
@@ -1,3 +1,26 @@
-// Add your code here
+const originalContentHtml = document.getElementById('content').innerHTML;
+let contentHtml = originalContentHtml;
+const input = document.getElementById('search-input');
 
-input.addEventListener('keydown', handleKeyDown);
+function handleKeyUp() {
+  contentHtml = originalContentHtml;
+  const searchWord = document.getElementById('search-input').value;
+
+  /**
+   * Regex and its boundary condition sourced from:
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/RegExp
+   * https://www.regular-expressions.info/wordboundaries.html
+   * https://stackoverflow.com/questions/17885855/use-dynamic-variable-string-as-regex-pattern-in-javascript
+   */
+  const wordRegex = new RegExp(`\\b${searchWord}\\b`, 'gi');
+  const innerHtml = `<span style = "background-color: #ffff00">${searchWord}</span>`;
+  contentHtml = contentHtml.replaceAll(wordRegex, innerHtml);
```

Comment:
Replacing with the search word itself will cause caps in the matching text to be lowercase to the user. Finding the fix for this takes a good amount of time, so it depends how important it is to you to fix it.